src/compiler/transformers/es2018.ts

```diff
@@ -713,6 +717,32 @@ export function transformES2018(context: TransformationContext): (x: SourceFile
         return result;
     }
 
+    /**
+     * Walk an AssignmentPattern to determine if it contains object rest (`...`) syntax. We cannot rely on
+     * propagation of `TransformFlags.ContainsObjectRestOrSpread` since it isn't propagated by default in
+     * ObjectLiteralExpression and ArrayLiteralExpression since we do not know whether they belong to an
+     * AssignmentPattern at the time the nodes are parsed.
+     */
+    function containsObjectRestOrSpread(node: AssignmentPattern): boolean {

```

Comment:
This function looks like a copy/paste of `propagateAssignmentPatternFlags` (I'm sure intentionally); is there any way we can share the code, or maybe run `propagateAssignmentPatternFlags` after the fact once we know how the nodes are being used?

Reply:
We don't want to mutate transform flags on existing nodes, though I can probably move `containsObjectRestOrSpread` out to `factory/utilities.ts` and have `propagateAssignmentPatternFlags` use it.

Reply:
That sounds good enough to me!